src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeeringStats/GetAzureRMExpressRouteCrossConnectionArpTableCommand.cs

```diff
@@ -0,0 +1,102 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Network.Models;
+using MNM = Microsoft.Azure.Management.Network.Models;
+using System.Linq;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Network.Version2017_10_01.Models;
+using Microsoft.Azure.Management.Network;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRMExpressRouteCrossConnectionArpTable"), OutputType(typeof(PSExpressRouteCircuitArpTable))]
+    public class GetAzureRMExpressRouteCrossConnectionArpTableCommand : ExpressRouteCrossConnectionBaseCmdlet
+    {
+        [Parameter(
+             Mandatory = true,
+             ValueFromPipelineByPropertyName = true,
+             HelpMessage = "The resource group name.",
+            ParameterSetName = "SpecifyByParameterValues")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Alias("Name", "ResourceName")]
+        [Parameter(
+             Mandatory = true,
+             ValueFromPipelineByPropertyName = true,
+             HelpMessage = "The Name of Express Route Cross Connection",
+            ParameterSetName = "SpecifyByParameterValues")]
+        [ValidateNotNullOrEmpty]
+        public string CrossConnectionName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Express Route Cross Connection",
+            ParameterSetName = "SpecifyByReference")]
+        [ValidateNotNullOrEmpty]
+        public PSExpressRouteCrossConnection ExpressRouteCrossConnection { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The PeeringType")]
+        [ValidateSet(
+           MNM.ExpressRoutePeeringType.AzurePrivatePeering,
+           MNM.ExpressRoutePeeringType.AzurePublicPeering,
+           MNM.ExpressRoutePeeringType.MicrosoftPeering,
+           IgnoreCase = true)]
+        public string PeeringType { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The DevicePath, can be either Primary or Secondary")]
+        [ValidateSet(
+            "Primary",
+            "Secondary",
+            IgnoreCase = true)]
+        public DevicePathEnum DevicePath { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            List<object> arpTables = null;
+            if (!string.IsNullOrWhiteSpace(CrossConnectionName))
```

Comment:
you don't really need this check... the param validation will make sure that this had a value that is not empty.

This comment applies to the other cmdlets as well

Reply:
Need to check whether the arguments have been specified by a cross connection reference or the Resource group name and resource name